### PR TITLE
add new agrument whether to remove core set from refined set

### PR DESCRIPTION
**Pull request by zhuhui-in**
I add a new function whether to remove core set from refined set because core set is included in the refine set totally in PDBBind.

### CODE REVIEW

**Comment by zhuhui-in** [comment]
Besides, I retrain PotentialNet on modified refined set and test on core set with random splitting method. The performance showed that the model has limited generalization ability.
refined_core

File: python/dgllife/data/pdbbind.py [hunk 1]
```diff
@@ -151,8 +154,23 @@ def _read_data_files(self, pdb_version, subset, load_binding_pocket):
...
+        if remove_coreset_from_refinedset:
+            if pdb_version == 'v2015':
+                with open(extracted_data_path + '/v2015/INDEX_core_data.2013','r') as f:
+                    for line in f:
+                        fields = line.strip().split()
+                        if fields[0] != "#" and fields[0] in pdbs:
+                            pdbs.remove(fields[0])
+
+            if pdb_version == 'v2007':
+                with open (extracted_data_path + '/2007/INDEX.2017.core.data','r') as f:
```
Comment (cyhFlight) [inline python/dgllife/data/pdbbind.py]: `'/2007/INDEX.2017.core.data'` should be `'/2007/INDEX.2007.core.data'`

File: python/dgllife/data/pdbbind.py [hunk 2]
```diff
@@ -151,8 +154,23 @@ def _read_data_files(self, pdb_version, subset, load_binding_pocket):
...
+                    for line in f:
+                        fields = line.strip().split()
+                        if fields[0] != "#" and fielsd[0] in pdbs:
+                            pdbs.remove(fields[0])
```
Comment (cyhFlight) [inline python/dgllife/data/pdbbind.py:173]: Fix `fielsd` for `fields`

File: python/dgllife/data/pdbbind.py [hunk 3]
```diff
@@ -78,19 +78,22 @@ class PDBBind(object):
...
+        Whether to remove core set from refined set when train with refined set and test with core set. 
+        Default to True.
     """
-    def __init__(self, subset, pdb_version, load_binding_pocket=True, sanitize=False, calc_charges=False,
-                 remove_hs=False, use_conformation=True,
+    def __init__(self, subset, pdb_version, load_binding_pocket=True, remove_coreset_from_refinedset=True, sanitize=False, 
+                 calc_charges=False, remove_hs=False, use_conformation=True,
                  construct_graph_and_featurize=ACNN_graph_construction_and_featurization,
-                 zero_padding=True, num_processes=None):
+                 zero_padding=True, num_processes=None, **kwargs):
```
Comment (cyhFlight) [inline python/dgllife/data/pdbbind.py]: Why do we need `**kwargs`?

**Review by cyhFlight — CHANGES_REQUESTED** [review]
Thanks very much for adding this feature. Fix the typos and I will merge the changes. Appreciate your efforts.

File: python/dgllife/data/pdbbind.py (hunk 3 above)
Comment (zhuhui-in) [reply python/dgllife/data/pdbbind.py]: not necessarily.

**Pull request closed without merging**